crates/bevy_sprite/src/render/mod.rs

```diff
@@ -160,25 +160,31 @@ impl SpritePipelineKey {
     const MSAA_MASK_BITS: u32 = 0b111;
     const MSAA_SHIFT_BITS: u32 = 32 - Self::MSAA_MASK_BITS.count_ones();
 
-    pub fn from_msaa_samples(msaa_samples: u32) -> Self {
+    #[inline]
+    pub const fn from_msaa_samples(msaa_samples: u32) -> Self {
         let msaa_bits =
             (msaa_samples.trailing_zeros() & Self::MSAA_MASK_BITS) << Self::MSAA_SHIFT_BITS;
-        Self::from_bits(msaa_bits).unwrap()
+        // SAFETY: The input is the correct type and the mask will ensure the output range is
+        // capped
+        unsafe { Self::from_bits_unchecked(msaa_bits) }
```

Comment:
Idk if this is a good enough reason to introduce unsafe code. Does const-ing this unlock specific scenarios we need? Are there performance benefits?

Reply:
```suggestion
        Self::from_bits_truncate(msaa_bits)
```
This should be `const`  and safe.

Reply:
Thanks, I overlooked that, will fix it shortly.

Reply:
Fixed.